File: webapp/src/components/next.rs
use crate::prelude::*;

#[derive(Properties, PartialEq)]
pub(crate) struct GDNextProps {
    pub display: Option<String>,
    pub url: String,
}

#[function_component(GDNext)]
pub(crate) fn gdnext(props: &GDNextProps) -> Html {
    let contexts = use_context::<Contexts>().expect("Contexts not found");
    let app_config = contexts.config.clone();
    let display = match props.display.to_owned() {
        Some(display) => display,
        None => match app_config.get_nav_from_path(&props.url, &contexts) {
            Some(nav) => nav.name,
            None => "Not Found".to_string(),
        },
    };
    let href = props.url.to_owned();
    html!(
        <>
            <Paper class="d-flex flex-column align-center justify-center mt-a">
                <Link class="f5" href="https://www.instagram.com/gasslerdesign">
                    <Avatar class="d-inline mr-3" icon="fa-brands fa-instagram" />
                    <span>{"Visit our Instagram"}</span>
                </Link>
            </Paper>
            <Paper class="d-flex flex-column align-center justify-center mt-10">
                <Link class="btn theme-info" {href}>
                    <span>{format!("Continue to {}", display)}</span>
                    <Avatar class="d-inline ml-2" icon="fa-duotone fa-right" />
                </Link>
            </Paper>
        </>
    )
}

File: webapp/wc/next.js
/* Display button to navigate to next page */
"use strict"
{
    class Next extends HTMLElement {
        constructor() {
            super();
            let t = this;
            t._insta = document.createElement("webui-flex")
            t._link = document.createElement("webui-flex");
            if (t.parentNode.nodeName === 'P') {
                let p = t.parentNode;
                t.parentNode.parentNode.insertBefore(t, p);
            }
        }
        static get observedAttributes() {
            return ['name', 'href'];
        }
        attributeChangedCallback(property, oldValue, newValue) {
            if (oldValue === newValue) return;
            if (newValue === null || newValue === undefined) {
                delete this[property];
            } else {
                this[property] = newValue;
            }
        }
        connectedCallback() {
            let t = this;
            if (!t.parentNode) { return; }
            t.parentNode.insertBefore(t._insta, t);
            t.parentNode.insertBefore(t._link, t);
            t._insta.innerHTML = webuiApplyAppData(`
<a href="https://www.instagram.com/gasslerdesign" class="f5">

<webui-avatar src="brands instagram"></webui-avatar>

Visit our Instagram

</a>
`);
            t._insta.classList.add('mt-a');
            t._insta.setAttribute('column', true);
            t._insta.setAttribute('align', 'center');
            t._insta.setAttribute('justify', 'center');
            t._link.innerHTML = webuiApplyAppData(`
<a href="${t.href}" class="btn theme-info">

Continue to ${t.name}

<webui-fa icon="right" family="duotone" class="mt-1"></webui-fa>

</a>
`);
            t._link.classList.add('mt-10');
            t._link.setAttribute('column', true);
            t._link.setAttribute('align', 'center');
            t._link.setAttribute('justify', 'center');
        }
        disconnectedCallback() { }
    }
    customElements.define('app-next', Next);
}